src/Basescape/PurchaseState.cpp

```diff
@@ -159,12 +175,39 @@ PurchaseState::PurchaseState(Base *base, CannotReequipState *parent) : _base(bas
 		_cats.push_back("STR_FILTER_MISSING");
 	}
 
-	RuleBaseFacilityFunctions providedBaseFunc = _base->getProvidedBaseFunc({});
+	auto providedBaseFunc = _base->getProvidedBaseFunc({});
+
+	// setup the filter methods to be used in standard situations (craft, soldiers, items).
+	constexpr auto costIsNotZero = [](const auto *rule) { return rule->getBuyCost() != 0; };
+	constexpr auto requirementsAreResearched = [](const auto *rule)
+	{
+		return _game->getSavedGame()->isResearched(rule->getRequirements());
+	};
+	auto necessaryBaseFunctionsPresent = [&providedBaseFunc](const auto *rule)
+	{
+		return (~providedBaseFunc & rule->getRequiresBuyBaseFunc()).none();
+	};
+	constexpr auto requiredCountryAllied = [](const auto *rule)
+	{
+		std::string_view requiredCountry = rule->getRequiresBuyCountry();
+		if (!requiredCountry.empty())
+		{
+			for (const auto &country : *_game->getSavedGame()->getCountries())
+			{
+				if (country->getRules()->getType() == requiredCountry)
+				{
+					return !country->getPact();
+				}
+			}
+		}
+		return true;
+	};
+
+	auto standardFilter = allOf(costIsNotZero, requirementsAreResearched, necessaryBaseFunctionsPresent, requiredCountryAllied);
```

Comment:
probably missing a function for item's "buy requirements"?
see below
or maybe something else was intended and forgotten...